### PR TITLE
Galaxykit Remotes, Distributions and Repositories - create and delete

### PR DESCRIPTION
**Pull request by MilanPospisil**
https://issues.redhat.com/browse/AAH-2243

### CODE REVIEW

File: galaxykit/command.py [hunk 1]
```diff
@@ -198,6 +201,54 @@ def report_error(resp):
             },
         },
     },
+    "remote": {
+        "help": "Remote repository",
```
Comment (himdel) [inline galaxykit/command.py:205]: a remote is not a repository, it's just sync config for a repository

File: galaxykit/command.py [hunk 2]
```diff
@@ -198,6 +201,54 @@ def report_error(resp):
...
+            "create": {
+                "args": {
+                    "name": {},
+                    "url": {},
+                }
+            },
+        },
+    },
+    "repository": {
+        "help": "Local repository",
```
Comment (himdel) [inline galaxykit/command.py:221]: repositories are both local and remote

File: galaxykit/command.py [hunk 3]
```diff
@@ -198,6 +201,54 @@ def report_error(resp):
...
+                "args": {
+                    "name": {},
+                    "--pipeline": {},
+                    "--remote": {},
+                }
+            },
+        },
+    },
+    "distribution": {
+        "help": "Distribution of local repository",
```
Comment (himdel) [inline galaxykit/command.py:238]: most repositories will have a distribution, not just local ones

**Comment by himdel** [comment]
The rest looks good :+1: :)

File: galaxykit/utils.py [hunk 4]
```diff
@@ -73,3 +73,7 @@ def wait_for_task(api_client, task, timeout=300, raise_on_error=False):
             ready = resp["state"] not in ("running", "waiting")
         time.sleep(5)
     return resp
+
+
+def parse_pulp_id(pulp_href):
+    return pulp_href.split("/")[-2]
```
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply galaxykit/utils.py:79]: You are right, that is totally obvious, I should read the last one, not the exact one. My mistake. This is example of using ChatGPT without much thinking - I asked it to parse last string separated by / and I gave it the example.

**Comment by himdel** [comment]
Tested in https://github.com/ansible/ansible-hub-ui/pull/3574 (https://github.com/ansible/ansible-hub-ui/actions/runs/4682109624/jobs/8295536826?pr=3574)

```
Error: Galaxykit failed: Traceback (most recent call last):
  File "/home/runner/.local/bin/galaxykit", line 8, in <module>
    sys.exit(main())
  File "/home/runner/.local/lib/python3.10/site-packages/galaxykit/command.py", line 723, in main
    resp = roles.delete_role(client, args.name)
  File "/home/runner/.local/lib/python3.10/site-packages/galaxykit/roles.py", line 69, in delete_role
    role_id = get_role_id(client, role_name)
  File "/home/runner/.local/lib/python3.10/site-packages/galaxykit/roles.py", line 27, in get_role_id
    return utils.pulp_href_to_id(resp["results"][0]["pulp_href"])
  File "/home/runner/.local/lib/python3.10/site-packages/galaxykit/utils.py", line 82, in pulp_href_to_id
    if re.match(uuid_regex, section):
NameError: name 're' is not defined
```

**Review by himdel — APPROVED** [review]

**Pull request merged**